src/utils/blockchain/math-helpers.ts

```diff
@@ -97,17 +97,10 @@ export const trimZero = (amount: string) => {
   return amount.replace(/([0-9]+(\.[0-9]+[1-9])?)(\.?0+$)/, '$1');
 };
 
-export const isValidNumerishValue = (value: any) => {
-  if (
-    value === Infinity ||
-    isNaN(Number(value)) ||
-    value === undefined ||
-    value === null
-  ) {
-    return false;
-  }
-  return true;
-};
+export const isValidNumerishValue = (value: any) =>
+  ['string', 'number'].includes(typeof value) &&
+  value !== '' &&
+  Number.isFinite(Number(value));
```

Comment:
Not sure if this PR is still needed as staking page appears fine on latest `development` branch, but in case it is - 

this was returning false when `value` is bignumber type (and therefore causing `toValidNumberishValue()` to always return 0), below code adds a check to handle those cases:
```suggestion
export const isValidNumerishValue = (value: any) =>
  (['string', 'number'].includes(typeof value) &&
    value !== '' &&
    Number.isFinite(Number(value))) ||
  (typeof value === 'object' && value.isFinite?.())
```